python/pyspark/sql/pandas/types.py

```diff
@@ -763,6 +764,9 @@ def _create_converter_from_pandas(
     error_on_duplicated_field_names : bool, optional
         Whether raise an exception when there are duplicated field names.
         (default ``True``)
+    ignore_unexpected_complex_type_values : bool, optional
+        Whether ignore the case where unexpected values are given for complex types.
```

Comment:
Can we add a bit more comment to explain what is an unexpected value for a complex type? I.e., provide some concrete examples.